File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int> freq;
        int n=nums.size();
        int maj=0;
        for(int x:nums){
            freq[x]++;
        }
        int majority=n/2;
        for(auto x : freq)
        {
            if(x.second>majority){
                maj=x.first;
            }
        }
        return maj;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> freq=new HashMap<>();
        for(int x:nums){
            freq.put(x,freq.getOrDefault(x,0)+1);
        }
        int highest=Integer.MIN_VALUE,element=0;
        for(Map.Entry<Integer,Integer> entry:freq.entrySet()){
            if(entry.getValue()>highest){
                highest=entry.getValue();
                element=entry.getKey();
            }
        }
        return element;
    }
}